src/telemetry/telemetry.c

```diff
@@ -757,6 +757,94 @@ add_replication_telemetry(JsonbParseState *state)
 #define REQ_RELS_CONTINUOUS_AGGS "continuous_aggregates"
 #define REQ_FUNCTIONS_USED "functions_used"
 #define REQ_REPLICATION "replication"
+#define REQ_ACCESS_METHODS "access_methods"
+
+/*
+ * Add the result of a query as a sub-object to the JSONB.
+ *
+ * Each row from the query generate a separate object keyed by one of the
+ * columns. Each row will be represented as an object and stored under the
+ * "key" column. For example, with this query:
+ *
+ *    select amname as name,
+ *           sum(relpages) as pages,
+ *			 count(*) as instances
+ *		from pg_class join pg_am on relam = pg_am.oid
+ *	  group by pg_am.oid;
+ *
+ * might generate the object
+ *
+ * {
+ *    "brin" : {
+ *       "instances" : 44,
+ *       "pages" : 432
+ *    },
+ *    "btree" : {
+ *       "instances" : 99,
+ *       "pages" : 1234
+ *    }
+ * }
+ */
+static void
+add_query_result_dict(JsonbParseState *state, const char *query)
+{
+	MemoryContext orig_context = CurrentMemoryContext;
+
+	int res;
+	if (SPI_connect() != SPI_OK_CONNECT)
+		elog(ERROR, "could not connect to SPI");
+
+	/* Lock down search_path */
+	res = SPI_execute("SET LOCAL search_path TO pg_catalog, pg_temp", false, 0);
+	if (res < 0)
+		ereport(ERROR, (errcode(ERRCODE_INTERNAL_ERROR), (errmsg("could not set search path"))));
```

Comment:
`ereport(ERROR, errcode(ERRCODE_INTERNAL_ERROR` is just elog, right? Maybe we should use `Ensure` for all internal errors instead.

Reply:
Good point. Switched.